src/main/java/com/example/square/validator/PositiveArrayValidator.java

```diff
@@ -0,0 +1,22 @@
+package com.example.square.validator;
+
+import com.example.square.model.PositiveArray;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class PositiveArrayValidator implements ConstraintValidator<PositiveArray, int[]> {
+    @Override
+    public boolean isValid(int[] value, ConstraintValidatorContext context) {
+        if (value == null) {
+            return true;
```

Comment:
Why is true for null  ?

Reply:
I thought that, because we don't need to check anything, so that nothing is 0 or negative when value is null. So checking process should end here, because assigning it to true, doesn't change anything. So, after that, I'm checking dimensions' length in getShape() method and if dimensions' length is 0, I throw exception there.